openlibrary/templates/lib/header_dropdown.html

```diff
@@ -12,25 +12,52 @@
 <div class="$(props['name'])-component header-dropdown">
     <details>
       <summary>
-        $if 'image' in props:
+        $if 'mask' in props and ctx.user:
+            <img class="account__icon" src="https://archive.org/services/img/$(get_internet_archive_id(ctx.user.key))" alt="My account">
+            <img class="header-dropdown__icon logged" src="$(props['image'])" alt="$(props['label'])"/>
+        $elif 'image' in props:
             <img class="header-dropdown__icon" src="$(props['image'])" alt="$(props['label'])"/>
         $elif 'label' in props:
             $(props['label'])
             <span class="shift">$_('Menu')</span>
 
         <img class="down-arrow" aria-hidden="true" src="/static/images/down-arrow.png" alt="" role="presentation" width="7" height="4">
       </summary>
-      <div class="$(props['name'])-dropdown-component navigation-dropdown-component">
+      $if 'mask' in props:
+        <div class="mask-menu"></div>
+      <div 
+      $if 'mask' in props:
+        class="app-drawer"
+      $else:
+        class="$(props['name'])-dropdown-component navigation-dropdown-component"
+      >
         <ul class="dropdown-menu $(props['name'])-dropdown-menu">
           $for link in props['links']:
-            <li>
-            $if 'post' in link:
-              <form action="$(link['post'])" method="post">
-                <button>$(link["text"])</button>
-              </form>
+            $if 'subsection' in link:
+              <li class="subsection">
+                $(link['subsection'])
+                $if link['subsection'] == 'My Open Library' and ctx.user:
+                  <img class="app-drawer__icon" src="https://archive.org/services/img/$(get_internet_archive_id(ctx.user.key))" alt="My account">
+              </li>
+            $elif 'loginClass' in link:
+              <li class="$(link['loginClass'])">
+                <a href="/account/login">$_('Log In')</a>
+                <a href="/account/create">$_('Sign Up')</a>
+              </li>
             $else:
-              <a href="$(link['href'])">$(link['text'])</a>
-            </li>
+              <li>
+              $if 'post' in link:
+                <form action="$(link['post'])" method="post">
+                  <button>$(link["text"])</button>
+                </form>
+              $else:
+                <a href="$(link['href'])">
+                  $if 'new' in link and props['name'] == 'hamburger':
+                    <span>$(link['text'])</span><span class="badge">NEW!</span>
```

Comment:
Make it uppercase with CSS.
```suggestion
                    <span>$(link['text'])</span><span class="badge">$_('New!')</span>
```